src/claimtrie.cpp

```diff
@@ -2527,79 +1756,107 @@ bool CClaimTrieCache::clear() const
     return true;
 }
 
+template <typename K, typename V>
+void CClaimTrieCache::update(std::map<K, V>& map)
+{
+    for (typename std::map<K, V>::iterator itQueue = map.begin(); itQueue != map.end(); ++itQueue) {
+        base->db.updateQueueRow(itQueue->first, itQueue->second);
+    }
+}
+
 bool CClaimTrieCache::flush()
 {
-    if (dirty())
+    if (dirty()) {
         getMerkleHash();
-    bool success = base->update(cache, cacheHashes, cacheTakeoverHeights, getBestBlock(), claimQueueCache, claimQueueNameCache, expirationQueueCache, nCurrentHeight, supportCache, supportQueueCache, supportQueueNameCache, supportExpirationQueueCache);
-    if (success)
-    {
-        success = clear();
     }
-    return success;
+
+    for (nodeCacheType::iterator itcache = cache.begin(); itcache != cache.end(); ++itcache) {
+        if (!base->updateName(itcache->first, itcache->second)) {
+            LogPrintf("%s: Failed to update name for:%s\n", __func__, itcache->first);
+            return false;
+        }
+    }
+    for (hashMapType::iterator ithash = cacheHashes.begin(); ithash != cacheHashes.end(); ++ithash) {
+        if (!base->updateHash(ithash->first, ithash->second)) {
+            LogPrintf("%s: Failed to update hash for:%s\n", __func__, ithash->first);
+            return false;
+        }
+    }
+    for (std::map<std::string, int>::iterator itheight = cacheTakeoverHeights.begin(); itheight != cacheTakeoverHeights.end(); ++itheight) {
+        if (!base->updateTakeoverHeight(itheight->first, itheight->second)) {
+            LogPrintf("%s: Failed to update takeover height for:%s\n", __func__, itheight->first);
+            return false;
+        }
+    }
+    update(claimQueueCache);
```

Comment:
Any way we can do these updates in a `for` loop instead? We have all the buffers in a collection, no?

Reply:
CClaimTrieCache::update has a for loop.